### PR TITLE
Add TxScript Codec

### PR DESCRIPTION
**Pull request by h0ngcha0**
As requested by the Mobula team

### CODE REVIEW

File: packages/web3/src/codec/txscript-codec.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+export class TxScriptCodec implements Codec<DecodedTxScript> {
+  parser = Parser.start().nest('methods', {
+    type: methodsCodec.parser
+  })
+
+  encode(input: DecodedTxScript): Buffer {
+    return methodsCodec.encode(input.methods.value)
+  }
```
Comment (Lbqds) [inline packages/web3/src/codec/txscript-codec.ts:40]: Does it make sense to add an `encodeTxScript(script: TxScript)` method?

**Review by Lbqds — APPROVED** [review]
LGTM 👍

File: packages/web3/src/codec/txscript-codec.ts (hunk 1 above)
Comment (h0ngcha0) [reply packages/web3/src/codec/txscript-codec.ts:41]: Good idea, added now.

File: packages/web3/src/codec/txscript-codec.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+export interface DecodedTxScript {
+  methods: DecodedArray<DecodedMethod>
+}
+
+export interface TxScript {
+  methods: Method[]
+}
+
+export class TxScriptCodec implements Codec<DecodedTxScript> {
```
Comment (polarker) [inline packages/web3/src/codec/txscript-codec.ts:34]: Is this used in transaction codec?

Comment (h0ngcha0) [reply packages/web3/src/codec/txscript-codec.ts:34]: Great catch, we already have a Script codec, i will merge these two.

**Pull request merged**